src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -0,0 +1,174 @@
+                CommandOptionType.SingleValue);
+
+            command._packageDirectoryOption = command.Option(
+                $"--package-directory <{LocalizableStrings.CmdPackageDirectory}>",
+                LocalizableStrings.CmdPackageDirectoryDescription,
+                CommandOptionType.SingleValue);
+
+            return command;
+        }
+
+        public override int Run(string fileOrDirectory)
+        {
+            var projects = new ProjectCollection();
+            var msbuildProj = MsbuildProject.FromFileOrDirectory(projects, fileOrDirectory);
+
+            if (RemainingArguments.Count == 0)
+            {
+                throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneReferenceToAdd);
+            }
+
+            var tempDgFilePath = string.Empty;
+            if(!_noRestoreOption.HasValue())
+            {
+                // Create a Dependency Graph file for the project
+                tempDgFilePath = CreateTemporaryFile(".dg");
+                GetProjectDependencyGraph(msbuildProj.ProjectRootElement.FullPath, tempDgFilePath);
+            }
+
+            var result = NuGetCommand.Run(TransformArgs(tempDgFilePath, msbuildProj.ProjectRootElement.FullPath));
+            DisposeTemporaryFile(tempDgFilePath);
+            return result;
+        }
+
+        private void GetProjectDependencyGraph(string projectFilePath,
+            string dgFilePath)
+        {
+            var args = new List<string>();
+
+            // Pass the project file path
+            args.Add(projectFilePath);
+
+            // Pass the task as generate restore Dependency Graph file
+            args.Add("/t:GenerateRestoreGraphFile");
+
+            // Pass Dependency Graph file output path
+            args.Add(string.Format("/p:RestoreGraphOutputPath={0}{1}{2}", '"', dgFilePath, '"'));
+
+            var result = new MSBuildForwardingApp(args).Execute();
+
+            if (result != 0)
+            {
+                throw new GracefulException(string.Format(LocalizableStrings.CmdDGFileException, projectFilePath));
+            }
+        }
+
+        private string CreateTemporaryFile(string extension)
+        {
+            var tempDirectory = Path.GetTempPath();
+            var tempFile = Path.Combine(tempDirectory, Guid.NewGuid().ToString() + extension);
+            File.Create(tempFile).Dispose();
+            return tempFile;
+        }
+
+        private void DisposeTemporaryFile(string filePath)
+        {
+            if (File.Exists(filePath))
+            {
+                File.Delete(filePath);
+            }
+        }
+
+        private string[] TransformArgs(string tempDgFilePath, string projectFilePath)
+        {
+            var args = new List<string>(){
+                "package",
+                "add",
+                "--package",
+                "Newtonsoft.Json",
+                "--project",
+                projectFilePath
+            };
```

Comment:
NIT: I usually prefer to put an empty line after a closing bracket. Of course, not in the if/else one.